src/reducerMaps/restaurants.js

```diff
@@ -265,10 +265,15 @@ export default new Map([
       items: {
         entities: {
           restaurants: {
-            [action.decision.restaurant_id]: {
-              all_decision_count: {
-                $apply: count => parseInt(count, 10) + 1
-              }
+            $apply: r => {
+              const changedRestaurants = Object.assign({}, r);
```

Comment:
`Object.assign()` doesn't do a deep copy, which means that while the `changedRestaurants` object is new, the individual restaurants are not. This means that components that might need to receive new versions of the restaurant, such as `RestaurantDropdown`, are receiving the same copy, albeit mutated. I can't actually explain while the component updates regardless of the restaurants being the same - it might be a bug with other props failing shallow comparison, although this could be investigated separately.

Anyway, this means that on line 270 and 273, you're mutating old restaurants. I would recommend _not_ returning a new `changedRestaurants` object, but rather assigning new objects to `r[action.decision.restaurant_id]` and `r[i.restaurant_id]`, with a new `all_decision_count` in both cases - something like this for the former:

```js
r[action.decision.restaurant_id] = {
  ...r[action.decision.restaurant_id],
  all_decision_count: parseInt(decision.all_decision_count, 10) + 1,
};
```